File: frontend/src/features/product_management_module/services/productService.js
import apiClient from "../../../common/api";

/**
 * Calls GET /api/v1/products (public, active products only).
 *
 * `keyword`/`categoryId` are optional — UC09 (Browse & Search Product).
 * Omit both (or call with no args) to get the full active catalog, same
 * as before search/filter existed.
 */
export async function browseProducts({ keyword, categoryId } = {}) {
  const params = {};
  if (keyword) params.keyword = keyword;
  if (categoryId) params.categoryId = categoryId;

  const { data } = await apiClient.get("/products", { params });
  return data.data;
}

/** Calls GET /api/v1/products/{productId}. */
export async function getProduct(productId) {
  const { data } = await apiClient.get(`/products/${productId}`);
  return data.data;
}

/** Manager catalog view: calls GET /api/v1/manager/products (includes inactive products). */
export async function listAllProductsForManager() {
  const { data } = await apiClient.get("/manager/products");
  return data.data;
}

/** Calls POST /api/v1/manager/products. */
export async function createProduct(payload) {
  const { data } = await apiClient.post("/manager/products", payload);
  return data.data;
}

/** Calls PUT /api/v1/manager/products/{productId}. */
export async function updateProduct(productId, payload) {
  const { data } = await apiClient.put(`/manager/products/${productId}`, payload);
  return data.data;
}

File: backend/src/main/java/com/freshmart/backend/service/interfaces/product_management_module/ProductService.java
package com.freshmart.backend.service.interfaces.product_management_module;

import java.util.List;

import com.freshmart.backend.dto.request.product_management_module.CreateProductRequest;
import com.freshmart.backend.dto.request.product_management_module.UpdateProductRequest;
import com.freshmart.backend.dto.response.product_management_module.ProductResponse;

/** Implements FE-03 (Product Management): customer browsing + manager CRUD. */
public interface ProductService {

    /**
     * UC09 - Browse & Search Product. keyword/categoryId are both optional
     * (pass null for either to skip that filter) — see
     * ProductRepository#searchActiveProducts.
     */
    List<ProductResponse> browseActiveProducts(String keyword, Long categoryId);

    /** Manager catalog view: every product regardless of isActive. */
    List<ProductResponse> listAllProducts();

    ProductResponse getProduct(Long productId);

    ProductResponse createProduct(CreateProductRequest request);

    ProductResponse updateProduct(Long productId, UpdateProductRequest request);
}
